### PR TITLE
Feat/delete user data

### PR DESCRIPTION
**Pull request by sebsch1337**
You can now delete your account :) please review it.

### CODE REVIEW

File: services/tripService.js [hunk 1]
```diff
@@ -57,3 +57,17 @@ export async function deleteTrip(id) {
 
   return deletedTrip;
 }
+
+export async function deleteUserAccount(userEmail) {
+  await dbConnect();
+
+  const trips = await Trip.find({ userEmail: userEmail });
+  if (!Array.isArray(trips)) throw new Error();
```
Comment (Kheirah) [inline services/tripService.js:65]: You can pass in a descriptive message inside of `Error()` for better developer experience (DX).

File: pages/details/[id].js [hunk 2]
```diff
@@ -51,6 +52,21 @@ export default function Details({ id, destinationDB, toDosDB }) {
 
   const toggleShowProfile = () => setShowProfile((showProfile) => !showProfile);
   const toggleLoader = () => setLoader((loader) => !loader);
+  const toggleModal = (modalName = "", type = "", id = "") =>
+    setModal((modal) => ({ visible: !modal.visible, name: modalName, type: type, id: id }));
```
Comment (Kheirah) [inline pages/details/[id].js:56]: I'm wondering, what do you need the `id` for here? I'm not critiquing, I'm just not seeing how the id is doing anything.

**Review by Kheirah — APPROVED** [review]
Good work!

Have a look at the comments; they're just minor things you may want to look into.

File: services/tripService.js (hunk 1 above)
Comment (sebsch1337) [reply services/tripService.js:65]: The error message (500 - internal server error) is thrown in the api catch. I throw this error here when there's an unexpected return (not array).

File: pages/details/[id].js (hunk 2 above)
Comment (sebsch1337) [reply pages/details/[id].js]: Man you have eagle eyes :) i removed the id everywhere since i never used it. Thanks!

**Pull request merged**